logbook-core/src/main/java/org/zalando/logbook/DefaultHttpLogFormatter.java

```diff
@@ -0,0 +1,131 @@
+package org.zalando.logbook;
+
+/*
+ * #%L
+ * Logbook
+ * %%
+ * Copyright (C) 2015 Zalando SE
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Multimaps;
+import lombok.SneakyThrows;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Multimaps.transformValues;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+public final class DefaultHttpLogFormatter implements HttpLogFormatter {
+
+    @Override
+    public String format(final String correlationId, final HttpRequest request) throws IOException {
+        final List<String> lines = new ArrayList<>();
+
+        lines.add(formatRequestLine(request));
+        lines.addAll(formatHeaders(request.getHeaders()));
+
+        final String body = request.getBodyAsString();
+        if (!body.isEmpty()) {
+            lines.add("");
+            lines.add(body);
+        }
+
+        return join(lines);
+    }
+
+    private String formatRequestLine(final HttpRequest request) {
+        return String.format("%s %s HTTP/1.1", request.getMethod(), getRequestURI(request));
```

Comment:
I changed it as follows:

- `HttpRequest.getRequestUri()` returns the original URI from the request line, including query paramters
- `HttpRequest.getParameters` no longer exists
- The parameter obfuscator works on query parameters alone
- There is no builtin support for form parameters in POST bodies (yet?)
- although that could be implemented using the existing body obfuscator interface and delegating to a key-value based obfuscator